src/shared/serverinfo.hpp

```diff
@@ -0,0 +1,231 @@
+#ifndef SERVERINFO_HPP
+#define SERVERINFO_HPP
+
+#include <vector>
+#include <string>
+
+#include "defs.hpp"
+
+struct serverinfo;
+
+// list of known servers, with or without informations
+// TODO:
+//   * should be a private static of serverinfo
+//   * stop using raw pointers (RAII ftw)
+extern vector<serverinfo*> servers;
+
+// code imported, here to reduce build dependencies
+extern bool filterstring(char *dst, const char *src, bool newline, bool colour, bool whitespace, bool wsstrip, size_t len);
+namespace client
```

Comment:
There should be a blank line above this line, like you have one below this block.